src/Response.php

```diff
@@ -139,13 +142,37 @@ public function resolveProperties(Request $request, array $props): array
         return $props;
     }
 
+    /**
+     * Resolve the ProvidesInertiaProps props.
+     */
+    public function resolveInertiaPropsProviders(array $props, Request $request): array
+    {
+        $newProps = [];
+
+        $renderContext = new RenderContext($this->component, $request);
+
+        foreach ($props as $key => $value) {
+            if (is_numeric($key) && $value instanceof ProvidesInertiaProps) {
+                // Pipe into a Collection to leverage Collection::getArrayableItems()
+                $newProps = array_merge(
+                    $newProps,
+                    collect($value->toInertiaProps($renderContext))->all()
```

Comment:
It would be helpful if you could call this using the container so we can inject it directly into the method.

```php
App::call([$value, 'toInertiaProps'], ['renderContext' => $renderContext])
```